src/main/java/io/pakland/mdas/githubstats/domain/Organization.java

```diff
@@ -32,4 +30,34 @@ public class Organization {
     orphanRemoval = true
   )
   private List<Team> teams = new ArrayList<>();
+
+  public void addTeam(Team team) {
+    if (teams == null) {
+      teams = new ArrayList<>();
+    }
+
+    if (!teams.contains(team)) {
+      teams.add(team);
+    }
+    team.setOrganization(this);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    Organization that = (Organization) o;
+
+    return id.equals(that.id);
```

Comment:
Regarding this, I get this warning:

> Using `@EqualsAndHashCode` for JPA entities is not recommended. It can cause severe performance and memory consumption issues.

Reply:
Also, about the `null` subject, we have the `Column` annotation with the `nullable = false`. Furthermore, I think we could ensure more the type-safetiness using the `@NotNull` annotation.

Reply:
> Regarding this, I get this warning:
> 
> > Using `@EqualsAndHashCode` for JPA entities is not recommended. It can cause severe performance and memory consumption issues.

Yes, you are right, didn't remember that. Then hardcoded better to avoid performance issues. Go ahead with your implementation, seems good enough. It has some issues with JPA state entity lifecycle, but I think we shouldn't worry right now, as that inconsistencies didn't apply to our app.

https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/

Reply:
> Also, about the `null` subject, we have the `Column` annotation with the `nullable = false`. Furthermore, I think we could ensure more the type-safetiness using the `@NotNull` annotation.

Good point. I was thinking that it is better to make an issue to make all validation regards the entities level